A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,442 @@
+
+Because xDS handles not only load balancing but also service discovery
+and configuration, gRPC will support xDS via both the resolver and LB
+policy plugins.  Both the resolver and LB policy plugins will need to
+communicate with the xDS server, so the functionality for interacting with
+the xDS server will be contained within an XdsClient object, which will
+be used by both the resolver and the LB policies.
+
+There will be two separate LB policies for xDS, one to support `Cluster`
+data, and another to support `ClusterLoadAssignment` data.
+
+Note that the LB policies will have "experimental" suffixes for their
+names for now.  These suffixes will be removed when the functionality
+has proven to be stable.
+
+![gRPC Client Architecture Diagram](A27_graphics/grpc_client_architecture.png)
+
+[Link to SVG file](A27_graphics/grpc_client_architecture.svg)
+
+#### XdsClient and Bootstrap File
+
+The XdsClient object contains all of the logic for interacting with the
+xDS server.  The XdsClient object is a purely internal API, not
+something exposed to gRPC users, but it is described here as part of
+the architecture used to support xDS in the gRPC client.
+
+The XdsClient object is configured via a bootstrap file.  The location
+of the bootstrap file is determined via the `GRPC_XDS_BOOTSTRAP`
+environment variable.  The file is in JSON form, and its contents look
+like this:
+
+```
+{
+  // The xDS server to talk to.  The value is an array to allow for a
+  // future change to add support for failing over to a secondary xDS server
+  // if the primary is down, but for now, only the first entry in the
+  // array will be used.
+  xds_servers": [
+    {
+      "server_uri": <string containing URI of xds server>,
+      // List of channel creds; client will stop at the first type it
+      // supports.  This field is optional; if not specified, xDS will use
+      // the same channel creds as the backends, but with any associated
+      // call creds stripped off.
+      "channel_creds": [
+        {
+          "type": <string containing channel cred type>,
+          // The "config" field is optional; it may be missing if the
+          // credential type does not require config parameters.
+          "config": <JSON object containing config for the type>
+        }
+      ]
+    }
+  ],
+  "node": <JSON form of Node proto>
+}
+```
+
+Initially, the only type of channel creds we support will be
+`google_default`.  In the future, we will add a general-purpose
+mechanism for configuring arbitrary channel creds types with arbitrary
+configuration.
+
+The `node` field will be populated with the [xDS `Node`
+proto](https://github.com/envoyproxy/data-plane-api/blob/1adb5d54abb0e28ca409254d26fad1cf5535239b/envoy/api/v2/core/base.proto#L85).
+Note that the `build_version`, `user_agent_name`, `user_agent_version`, and
+`client_features` fields should not be specified in the bootstrap file, since
+they will be populated automatically by the gRPC xDS client.
```

Comment:
Will gRPC announce the enhanced `Node` structure with user name/versioning information as per https://github.com/envoyproxy/envoy/pull/9301/files#diff-ebbcbb34815ee818f64401154d5b1de6R161?